samples: net: wpanusb: cleanup configs and documentation

mike-scott opened the pull request:
This PR does the following:
- Removes HW specific settings from the default prj.conf.  This allows HW with native support for 802.15.4 to be built by default (nrf52840_pca10056 and usb_kw24d512).
- Split out CC2520 settings into an overlay config
- Add overlay configs for CC1200 and MCR20A
- Update documentation to rst format and add more instructions

mike-scott committed:
samples: net: wpanusb: split cc2520 settings from prj.conf

When the CC2520-specific configs are removed from the prj.conf
file, the sample can be built for hardware that supports
802.15.4 natively w/o any changes to the prj.conf like so:
cmake -DBOARD=nrf52840_pca10056 ..
cmake -DBOARD=usb_kw24d512 ..

The CC2520-specific settings now live in overlay-cc2520.conf
which can be used like so when building for devices that can
use it:
cmake -DBOARD=quark_se_c1000_devboard \
  -DOVERLAY_CONFIG=overlay-cc2520.conf ..

Later, support for other optional HW like CC1200 and MCR20A
can be added in a similar way.

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott committed:
samples: net: wpanusb: add TI CC1200 overlay config

Add support for TI CC1200 802.15.4 module as an overlay config file.

Use this file like so:
cmake -DBOARD=quark_se_c1000_devboard \
      -DOVERLAY_CONFIG=overlay-cc1200.conf ..

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott committed:
samples: usb: wpanusb: add MCR20A overlay config

Add support for NXP MCR20A 802.15.4 module as an overlay config file.

Use this file like so:
cmake -DBOARD=usb_kw24d512 \
      -DOVERLAY_CONFIG=overlay-mcr20a.conf ..

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott committed:
samples: usb: wpanusb: update README to rst format

Let's rename README to README.rst and add:
- Some nice formatting for existing content
- Requirements section (including handy links)
- Build and Running section (including multiple overlay options)

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott commented:
Fixed 1 config bug and a typo in a commit description.

mike-scott commented:
Looks like a change to samples.yaml is also in order now that the default build works for nrf52840_pca10056 and usb_kw24d512

mike-scott commented:
Fixed a documentation check fail.

finikorg reviewed (approved):
LGTM

pfalcon commented on samples/net/wpanusb/README.rst:
```diff
@@ -0,0 +1,92 @@
+Requirements
+************
+
+- Linux machine host connected via USB to Zephyr WPANUSB enabled MCU
```
"Linux machine host" sounds a bit artificial. Maybe just "Linux host"?

pfalcon replied:
"Zephyr WPANUSB enabled MCU" sounds strange. Maybe "Zephyr WPANUSB enabled board"?

pfalcon replied:
Also, we usually consider smaller boards to be connected to a host, not vice-versa ;-).

pfalcon reviewed (approved):
Some suggestions to clarify docs.

jfischer-phytec-iot commented on samples/net/wpanusb/README.rst:
```diff
@@ -0,0 +1,92 @@
+   ...
+   T:  Bus=01 Lev=02 Prnt=02 Port=00 Cnt=01 Dev#=  3 Spd=12  MxCh= 0
+   D:  Ver= 1.10 Cls=ff(vend.) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
+   P:  Vendor=8086 ProdID=ff03 Rev=01.00
```
It is no longer Intel VID/PID, please change this line to `P:  Vendor=2fe3 ProdID=0101 Rev=01.00`

mike-scott replied:
Nice catch.  Fixing.

dbkinder commented on samples/net/wpanusb/README.rst:
```diff
@@ -0,0 +1,92 @@
+********
+
+This application exports ieee802154 radio over USB to be used in other
+Operation Systems like Linux. In this scenario Linux SoftMAC driver would
```
change to:

    This application exports ieee802154 radio over USB to be used in
    other OSes such as Linux.  For Linux, the ieee802154 stack would be
    implemented using the Linux SoftMAC driver.

dbkinder commented on samples/net/wpanusb/README.rst:
```diff
@@ -0,0 +1,92 @@
+Requirements
+************
+
+- Linux machine host connected via USB to Zephyr WPANUSB enabled MCU
```
How about this:

    - a Zephyr WPANUSB enabled board (such as the :ref:`nrf52840_pca10056`)
      connected via USB to a Linux host.

dbkinder commented on samples/net/wpanusb/README.rst:
```diff
@@ -0,0 +1,92 @@
+********************
+
+There are configuration files for various setups in the
+samples/net/wpanusb directory:
```
change to

    ``samples/net/wpanusb`` directory:

dbkinder commented on samples/net/wpanusb/README.rst:
```diff
@@ -0,0 +1,92 @@
+samples/net/wpanusb directory:
+
+- :file:`prj.conf`
+  This is the standard default config.  This can be used by itself for
```
To have this displayed nicely, indent the "body" under these file names by two spaces:

    :file:``prj.conf``
      This is the standard default config.  This can be used by itself for
      hardware which has native 802.15.4 support.

    :file:`overlay-cc1200.conf`
      This overlay config enables support for CC1200

and so on.

dbkinder commented on samples/net/wpanusb/README.rst:
```diff
@@ -0,0 +1,92 @@
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/wpanusb
+   :host-os: unix
+   :board: quark_se_c1000_devboard
```
I don't see this as a listed Zephyr supported board in our list http://docs.zephyrproject.org/boards/x86/index.html

mike-scott replied:
Interesting, it was the only board which had settings in place for CC2520.  So I figured it was used for testing.  https://github.com/zephyrproject-rtos/zephyr/tree/master/boards/x86/quark_se_c1000_devboard

mike-scott committed:
samples: net: wpanusb: fix sanitycheck

With the change to the configuration files, we can now compile
this sample for any platform that uses only the default prj.conf file:
nrf52840_pca10056
nrf52840_pca10059
usb_kw24d512

Update the sample.yaml file to remove the excludes.

Signed-off-by: Michael Scott <mike@foundries.io>

carlescufi commented:
@dbkinder can you revisit this?

lemrey reviewed (commented):
@mike-scott Looks good to me, the example compiles for the nrf52840_pca10056 (and the upcoming nrf52840_pca10059). Perhaps the board could be removed the whitelist?

jfischer-phytec-iot commented on samples/net/wpanusb/overlay-mcr20a.conf:
```diff
@@ -0,0 +1,6 @@
+CONFIG_GPIO=y
+CONFIG_SPI=y
+CONFIG_SPI_0=y
```
The build for frdm_k64f will fail. With the consideration of e240773d166081aa1a18b2d3d4ba3645873d1622 the case as it is here remains untested. I recommend to change the example in the commit message from frdm_k64f to usb_kw24d512. The line 3 must then be removed.

mike-scott commented:
@lemrey to clarify, are you saying we should just remove the whitelist?

mike-scott commented:
Changes:
- Removed extra 3 config lines from MCR20A overlay and changed commit description to use usb_kw24d512 in the example (instead of K64F)
- Removed the whitelist in sample.yaml.   We should be able to run the default boards for sanitycheck.

Diff from V2: https://hastebin.com/soqaqizoju.diff

finikorg commented:
@mike-scott after you have removed whitelist should sanitycheck build sample for all supported boards? It does not work for me with `sanitycheck -t usb`

mike-scott commented:
@finikorg Hm.  I wasn't seeing any sanitycheck errors (and the run for this PR didn't fail).  Did https://github.com/zephyrproject-rtos/zephyr/pull/8984 fix the issue?  Which board combination was failing?

finikorg commented:
@mike-scott There is no error, after whitelist is removed the test is not built at all. You can check `sanity-out/make.log`.

jfischer-phytec-iot commented:
The testing shoud be turned on for usb_kw24d512 or nrf52840_pca10056.

```
--- a/boards/arm/nrf52840_pca10056/nrf52840_pca10056.yaml
+++ b/boards/arm/nrf52840_pca10056/nrf52840_pca10056.yaml
@@ -9,3 +9,5 @@ supported:
   - usb_device
   - ble
   - ieee802154
+testing:
+  default: true

--- a/boards/arm/usb_kw24d512/usb_kw24d512.yaml
+++ b/boards/arm/usb_kw24d512/usb_kw24d512.yaml
@@ -11,3 +11,5 @@ supported:
   - ieee802154
   - usb_device
   - watchdog
+testing:
+  default: true

```

mike-scott commented:
@jfischer-phytec-iot @finikorg I guess the question is: do we want to enable default testing on these platforms?

finikorg commented:
@mike-scott I do not think all (similar) boards needs to be added, my point was that this test was not triggered at all. Now I have enabled quark_se platform with PR #9051 so this test runs at least once.
